matrix-sdk-android/src/main/java/org/matrix/android/sdk/api/util/MatrixItem.kt

```diff
@@ -37,11 +38,20 @@ sealed class MatrixItem(
                         override val displayName: String? = null,
                         override val avatarUrl: String? = null) :
             MatrixItem(id, displayName?.removeSuffix(IRC_PATTERN), avatarUrl) {
+
         init {
             if (BuildConfig.DEBUG) checkId()
         }
 
         override fun updateAvatar(newAvatar: String?) = copy(avatarUrl = newAvatar)
+
+        companion object {
+            fun fromJson(userId: String, profileJson: JsonDict) = UserItem(
```

Comment:
I would prefer that this does not exist.

You can use instead `User.fromJson(...).toMatrixItem()`

But `MatrixItem` is more something that the SDK expose to the app. The SDK should not use `MatrixItem` internally. You should use `User` instead.